model-optimizer/extensions/ops/If.py

```diff
@@ -0,0 +1,308 @@
+        }
+        base_attrs.update(attrs)
+        super().__init__(graph, base_attrs, attrs)
+
+    def port_map_attrs(self):
+        return [
+            'external_port_id',
+            'internal_layer_id'
+        ]
+
+    @staticmethod
+    def connect_body_input(if_node: Node, condition: bool, if_input_port_idx: int, body_parameter: Node):
+        """
+        Update the input port map to connect the input port with the specified body parameter
+
+        :param if_node: the If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :param if_input_port_idx: the input port index to connect
+        :param body_parameter: the body parameter node to connect
+        :return: None
+        """
+        assert if_node.soft_get('op') == 'If'
+        assert body_parameter.soft_get('op') == 'Parameter'
+        sub_graph = if_node.then_graph if condition else if_node.else_graph
+        assert body_parameter.id in sub_graph
+        body_parameter['input_id'] = if_input_port_idx
+
+    @staticmethod
+    def connect_body_output(if_node: Node, condition: bool, if_output_port_idx: int, internal_result: Node):
+        """
+        Update the output port map to connect the body Result node with the specified output port
+
+        :param if_node: the If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :param if_output_port_idx: the output port index to connect
+        :param internal_result: the body Result node to connect
+        :return: None
+        """
+        assert if_node.soft_get('op') == 'If'
+        assert internal_result.soft_get('op') == 'Result'
+        sub_graph = if_node.then_graph if condition else if_node.else_graph
+        assert internal_result.id in sub_graph
+        internal_result['output_id'] = if_output_port_idx
+
+    @staticmethod
+    def update_body_parameters_type(if_node: Node, condition: bool):
+        """
+        Update the data type for If body Parameter nodes based on data type of the outer graph nodes producing data
+        for them.
+
+        :param if_node: The If node
+        :param condition: the boolean defining a condition (then/else) graph
+        :return: None
+        """
+        assert if_node.soft_get('type') == 'If'
+
+        subgraph = if_node.then_graph if condition else if_node.else_graph
+        for node in subgraph.get_op_nodes():
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_type = if_node.in_port(input_port_id).get_data_type()
+                node.data_type = input_type
+                log.debug('Updated data type for the body node with name "{}" with value {}'
+                          .format(node.name, node.data_type))
+
+    @staticmethod
+    def updated_body_parameters_shape(if_node: Node, condition: bool):
```

Comment:
Why we need the second parameter? Could we just update shapes for both then and else?